src/webapp/components/settings-dialog/useLocaleSelector.ts

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { useSnackbar } from "@eyeseetea/d2-ui-components";
+import { useAppContext } from "$/webapp/contexts/app-context";
+import { LoadingState } from "$/webapp/pages/landing/useDataSetSelector";
+import { Locale } from "$/domain/entities/Locale";
+import i18n from "$/utils/i18n";
+import _ from "$/domain/entities/generic/Collection";
+
+export function useLocaleSelector() {
+    const { compositionRoot } = useAppContext();
+
+    const snackbar = useSnackbar();
+
+    const [locales, setLocales] = React.useState<Locale[]>([]);
+    const [loading, setLoading] = React.useState<LoadingState>("loading");
+    const [selected, setSelected] = React.useState<string>("en");
+
+    const items = React.useMemo(
+        () =>
+            locales.map(locale => ({
+                value: locale.code,
+                text: locale.displayName,
+            })),
+        [locales]
+    );
+
+    const onChange = React.useCallback((value: string) => {
+        setSelected(value);
+    }, []);
+
+    const props = React.useMemo(
+        () => ({
+            items: items,
+            value: selected,
+            onChange: onChange,
+            name: "select-language",
+            label: i18n.t("Language"),
+            disabled: loading === "loading",
+        }),
+        [items, selected, onChange, loading]
+    );
+
+    React.useEffect(
+        () =>
+            compositionRoot.locales.get.execute().run(
```

Comment:
Remember always to `return` the result of `run`, which is a cancel function (what useEffect needs if the component has be to be cleaned up)

Reply:
Ohh now I just truly understand the need of the `return` of the run 🙌🏼 was not thinking about in the case the component is cleaned up the request is no longer needed to keep processing 🥲